### ISSUE TITLE
Question: Shouldn't this process involve the server side?

### ISSUE THREAD

**Issue by raymondcarl** [xcv58/meteor-auth0#19]
Hi,

First, thank you for creating this package because I didn't really find the tutorials on the web helpful and having a package like this is really helpful.

I finally got it to work after having some issues with trying to redirect to localhost:3000.  But, now I'm reviewing how this flow works.  I know that Github issues are not the place for questions, but for lack of a better place because of how specific it is.... The whole process seems to run on the client side.  Are you not limited by what you can do in terms of limiting server to client data flows?  

I was trying to implement something like this, but most of the tutorials center around node.js.
![image](https://user-images.githubusercontent.com/9866425/43163815-9c77a5d8-8f8f-11e8-89a1-0d51e3705476.png)

By running everything on the client side, how does the server "trust" the token from the client side? 

Apologies if none of this makes any sense.....

**Comment by raymondcarl** [comment]
Just reading more into this and thought I saw something that helps.  Basically, you validate the token from the clientside kind of like this?
https://auth0.com/docs/tokens/id-token

**Comment by xcv58** [comment]
It does validate the `accessToken` on the server side: 

https://github.com/xcv58/meteor-auth0/blob/8894fa9922810e98809de9c77f583e8da3453756/auth0-lock/auth0-lock-server.js#L38

**Comment by xcv58** [comment]
No. Can you explain what's `when I add on SMS`? Is there any doc?

**Comment by raymondcarl** [comment]
By the way, does this apply to the package?

https://community.auth0.com/t/auth0-legacy-lock-api-disabled-grace-period-available/12949

> On Monday, July 16, 2018, Auth0 will begin disabling the deprecated Legacy Lock API via a staged rollout to all cloud tenants. This breaking change mitigates the vulnerability disclosed in April 2018 15 by preventing the /usernamepassword/login and /ssodata endpoints from being used or exploited in embedded mode. The Legacy Lock API will be disabled in all cloud tenants on or before Sunday, July 22, 2018.
> 
> Any applications that have not been fully migrated may experience outages and failed logins as a result of this change. Those that have been fully migrated to Universal Login or to the most recent versions of Lock (11+) and/or Auth0.js (9+) should continue to function normally. However, any direct calls to the disabled endpoints will fail when service is removed.

**Comment by xcv58** [comment]
We are using Lock 11+ https://github.com/xcv58/meteor-auth0/blob/8894fa9922810e98809de9c77f583e8da3453756/auth0-lock/package.js#L13

**Comment by raymondcarl** [comment]
It's just adding SMS for the multifactor authenteication.  So you authenticate via whichever connection (Database, google, etc.) and then you get an SMS code.  I enter the SMS code correctly, it accepts it as correct, but then I get referred to https://[issuer]/mf?provider=guardian&state=4ylv93qoPRMVwwHMajWV00gmMP4V0gf2

https://auth0.com/docs/multifactor-authentication

**Comment by xcv58** [comment]
What's `[issuer]`?

**Comment by raymondcarl** [comment]
Oh sorry, that is the Auth0 URL/domain used for your specific application.

**Comment by xcv58** [comment]
The callback URL should be handled by `Auth0Lock`. Could you double check whether `Auth0Lock` supports multifactor authentication?

https://github.com/xcv58/meteor-auth0/blob/8894fa9922810e98809de9c77f583e8da3453756/auth0-lock/auth0-lock-client.js#L44

**Comment by raymondcarl** [comment]
OK.  I checked the NPM package repo, and it looks like someone else had the same issue.  I guess you can, but it doesn't seem like a great option to enable CORS for the authentication process.  It seems like you have to eventually move to a hosted login page if you want to stay with Auth0 over the long term.
